src/main/java/core/basesyntax/service/CalculateBalanceService.java

```diff
@@ -0,0 +1,9 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+import java.util.Map;
+
+public interface CalculateBalanceService {
```

Comment:
You should have just one report generator, and it should deal with Storage/Dao directly, why would there be arguments?
review all your methods that deal with Map<String,Integer> and check that you really need them or is it over-engineering